packages/edit-site/src/components/save-button/index.js

```diff
@@ -6,13 +6,18 @@ import { Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { store as coreStore } from '@wordpress/core-data';
 import { displayShortcut } from '@wordpress/keycodes';
+import { useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { store as editSiteStore } from '../../store';
 
 export default function SaveButton() {
+	// We only want to show the saving state of the button if the button itself initiated the save,
+	// otherwise, the button will show in the side bar when saving is initiated from template save models, etc.
+	const [ selfInitiatedSave, setSelfInitiatedSave ] = useState( false );
```

Comment:
I'm not sure if this is the _right_ approach, as we're relying on the parent component to "remount" the save button to reset the state. At the time, it seems to be the case in the sidebar, as we'll not mount the button if `isDirty` is false. However, this breaks the editor that it'll not show the save button when first entering the editor.

Reply:
Seems like a lot of the `template-revert` e2e tests are failing randomly because the save button is sometimes not present 😅 .

Reply:
Yeh, I thought this was a flakey fix, so I am glad it made the tests flakey to prove that 😆 I will take another look at some different approaches, let me know if you have any ideas.